pkg/resources/jobs/runner.go

```diff
@@ -24,9 +24,16 @@ type Script struct {
 // NewRunnerJob creates a new k6 job from a CRD
 func NewRunnerJob(k6 *v1alpha1.K6, index int) (*batchv1.Job, error) {
 	name := fmt.Sprintf("%s-%d", k6.Name, index)
-	postCommand := []string{"k6", "run"}
+	command := []string{"k6", "run"}
+	image := "ghcr.io/grafana/operator:latest-runner"
 
-	command, istioEnabled := newIstioCommand(k6.Spec.Scuttle.Enabled, postCommand)
+	if k6.Spec.ServiceMesh.Istio.Enabled {
+		command = newIstioCommand(k6.Spec.ServiceMesh.Istio.Enabled, command)
+		image = "ghcr.io/grafana/operator:latest-runner-istio"
+	} else if k6.Spec.ServiceMesh.Linkerd.Enabled {
+		command = newLinkerdCommand(k6.Spec.ServiceMesh.Linkerd.Enabled, command)
+		image = "ghcr.io/grafana/operator:latest-runner-linkerd"
+	}
```

Comment:
A change request. Could you please pull this out into a separate type + function so that here in runner we'll have the following:

```golang
var base = BaseContainer{
  Command: []string{"k6", "run"},
  Image: "ghcr.io/grafana/operator:latest-runner",
  Env: k6.Spec.Runner.Env,
}

base.Adjust(k6.Spec.ServiceMesh)
...
```
And then in a separate go file `basecontainer.go` smth like:
```golang
// BaseContainer contains min of info about container that might be modified when mesh service is used
type BaseContainer struct {
  Command []string
  Image string
  Env []corev1.EnvVar
}

func (bs *BaseContainer) Adjust(mesh v1alpha1.K6ServiceMesh) {
    if mesh.Istio.Enabled {
        bs.Command = newIstioCommand(mesh.Istio.Enabled, bs.Command)
	bs.Image = fmt.Sprintf("%s-istio", bs.Image)
        bs.Env = append(bs.Env, newIstioEnvVar(mesh.Istio, mesh.Istio.Enabled))
    } else if mesh.Linkerd.Enabled {
	bs.Command = newLinkerdCommand(mesh.Linkerd.Enabled, command)
	bs.Image = fmt.Sprintf("%s-linkerd", bs.Image)
    }
    // no mesh - nothing to adjust
}
```
Basically, the goal is to keep all of these `if`s around meshes in one place. It might lead to simplifying some of the `newIstio*` and `newLinkerd*` functions too